src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -43,6 +46,28 @@ public static BooleanExpression needsStudyDatabase(StateManager stateManager) {
         return BooleanExpression.booleanExpression(binding);
     }
 
+    // Makes sure there is at least one group selected, and if there are multiple groups selected
+    // all have the same parent node.
+    public static BooleanExpression needsSelectedGroupsShareParent(StateManager stateManager) {
+        if (stateManager.activeGroupProperty().isEmpty()) {
+            return Bindings.size(stateManager.activeGroupProperty()).isEqualTo(1);
+        }
+        return Bindings.size(stateManager.activeGroupProperty()).isEqualTo(1)
+           .or(Bindings.size(stateManager.activeGroupProperty()).greaterThan(1).and(
+                Bindings.size(
+                        (ObservableList<GroupTreeNode>) stateManager
+                                .activeGroupProperty()
+                                .stream()
+                                .filter(stateManager.activeGroupProperty().get().getFirst()::equals)
+                                .collect(Collectors.toCollection(FXCollections::observableArrayList))
```

Comment:
Reads strange. This is an active query. Can't one have a "passive" list build dynamically?

The "dynamics" of bindings are explained well here: https://docs.oracle.com/javafx/2/binding/jfxpub-binding.htm

Reply:
I believe this should fix the problem, and get rid of the if-statement.
```
    public static BooleanExpression needsSelectedGroupsShareParent(StateManager stateManager) {
        return Bindings.size(stateManager.activeGroupProperty()).isEqualTo(1)        // Is single element OR
            .or(Bindings.size(stateManager.activeGroupProperty()).greaterThan(1)     // (Is multiple elements AND
                .and(Bindings.createBooleanBinding(
                    () -> stateManager.activeGroupProperty().stream()
                        .allMatch(val ->                                                // The parent of the first element
                            stateManager.activeGroupProperty()                          // is equal to the parent of all
                                .getFirst()                                             // other elements 
                                .getParent()                                            // (including if they are null))
                                    .map(node -> node.equals(val.getParent().get()))
                                    .orElseGet(() -> val.getParent().isEmpty())),
                stateManager.activeGroupProperty())));
    }
```